akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -249,16 +251,28 @@ private[projection] class R2dbcOffsetStore(
   private val selectOffsetSql: String =
     sql"SELECT projection_key, current_offset, manifest, mergeable FROM $offsetTable WHERE projection_name = ?"
 
-  private val upsertOffsetSql: String = sql"""
-    INSERT INTO $offsetTable
-    (projection_name, projection_key, current_offset, manifest, mergeable, last_updated)
-    VALUES (?,?,?,?,?,?)
-    ON CONFLICT (projection_name, projection_key)
-    DO UPDATE SET
-    current_offset = excluded.current_offset,
-    manifest = excluded.manifest,
-    mergeable = excluded.mergeable,
-    last_updated = excluded.last_updated"""
+  private val upsertOffsetSql: String = {
+    if (dialectName == "h2") {
+      // FIXME is branching good enough for now?
```

Comment:
um, maybe we should prepare for dialects and dao classes in similar way as the r2dbc plugin. When we add another it will always add to Projections too.

Reply:
or let the Dialect create the R2dbcOffsetStore and then only minimal overrides?

Reply:
This is pretty much what I ended up doing.